src/lib/prover/prover.ml

```diff
@@ -35,6 +36,15 @@ module type S = sig
     -> Consensus.Data.Prover_state.t
     -> Pending_coinbase_witness.t
     -> Blockchain.t Deferred.Or_error.t
+
+  val prove :
+       t
```

Comment:
We basically define S in this file and its corresponding mli file. Can we define S in its own separate intf.ml file and share them between the two files.